lib/active_model/serializer.rb

```diff
@@ -217,7 +217,7 @@ def each_association(&block)
         if serializer_class
           serializer = serializer_class.new(
             association_value,
-            options.merge(serializer_from_options(association_options))
+            options.except(:serializer).merge(serializer_from_options(association_options))
```

Comment:
This is the simplest thing that worked without doing a bunch of refactoring. (We considered adding a `::new_association_serializer` that wraps `::new` but it wasn't a quick win.)